### PR TITLE
add .npmrc with package-lock=false to disable package-lock.json generation

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (climba03003) [inline .npmrc:1]: Why it is needed? and what does it solve?

Comment (climba03003) [reply .npmrc:1]: I means why it need to disable the lock file generation.
If it is something specific to your environment or personal perference, then it shouldn't be checked in git repository.

Comment (Uzlopak) [reply .npmrc:1]: Main Repo has the .npmrc also
https://github.com/fastify/fastify/blob/main/.npmrc

Comment (climba03003) [reply .npmrc:1]: That's not a reason.
Here is the PR adding `.npmrc` with other files in `fastify`. 
https://github.com/fastify/fastify/pull/2269

But, it didn't explain why disable the `package-lock.json` improve DX.
We can use `npm udpate` if we install the dependency once.

Comment (Uzlopak) [reply .npmrc:1]: npm update does not depend on the lockfile.

@jsumners Any argument why disabling the unnecessary generation of the package-lock.json is a good idea?

Comment (climba03003) [reply .npmrc:1]: Disable `package-lock.json` means every run for `npm install` should fetch and install the latest.
Which means after the first `npm install`, it is actually doing the same thing as `npm update`.

**Review by jsumners — APPROVED** [review]

File: .npmrc (hunk 1 above)
Comment (jsumners) [reply .npmrc:1]: There is not point to having a package lock in a library project. They are ignored when installing the module as a dependency. The only thing having a package lock in this project will accomplish is to lock versions across contributors to the time when the various contributors start their work on the project.

**Pull request merged**